### PR TITLE
introduce Http retry strategy

### PR DESCRIPTION
**Pull request by jupe**
this PR introduce retry logic if uri that provides `resources.json` fails. Rules for retry are:
* total=5 attempts,
* follow redirect 5 times
* rely in following statuses:
  ```
   429,  # Too Many Requests
   500,  # Internal Server Error
   502,  # Bad Gateway server error
   503,  # Service Unavailable
   504  # Gateway Timeout server error
  ```
* backoff_factor=0.5

### CODE REVIEW

File: lockable/provider_http.py [hunk 1]
```diff
@@ -23,8 +53,7 @@ def reload(self) -> None:
     def _get_http(uri: str) -> list:
```
Comment (juhhov) [inline lockable/provider_http.py:53]: Missing self parameter.

**Comment by juhhov** [comment]
Missing tests, otherwise looks good.

File: lockable/provider_http.py [hunk 2]
```diff
@@ -13,18 +16,43 @@ class ProviderHttp(Provider):
...
         """ ProviderHttp constructor """
+        self._configure_http_strategy(uri)
         super().__init__(uri)
 
+    def _configure_http_strategy(self, uri):
+        """ configure http Strategy """
+        retry_strategy = Retry(
+            total=5,
+            redirect=5,
+            status_forcelist=[
```
Comment (juhhov) [inline lockable/provider_http.py:27]: What about the case where there is no server at all?

File: lockable/provider_http.py [hunk 3]
```diff
@@ -13,18 +16,43 @@ class ProviderHttp(Provider):
 
     def __init__(self, uri: str):
         """ ProviderHttp constructor """
+        self._configure_http_strategy(uri)
         super().__init__(uri)
 
+    def _configure_http_strategy(self, uri):
+        """ configure http Strategy """
+        retry_strategy = Retry(
```
Comment (juhhov) [inline lockable/provider_http.py:24]: How long is total retry duration?

Comment (jupe) [reply lockable/provider_http.py:24]: calculation: `{backoff factor} * (2 ** ({number of total retries} - 1))`
with 5 retries and backoff_factor 0.5, retries happens `[0.0s, 0.5s, 1.0s, 2.0s, 4.0s].

I increase retries to 9 and `backoff_factor` to 1, so last retry happens after 2min8s  - does it sounds enough even with short service breaks ?

File: lockable/provider_http.py (hunk 2 above)
Comment (jupe) [reply lockable/provider_http.py:35]: added `connect` parameter that retries connection errors.

**Comment by jupe** [comment]
There might be some cases still that is not retries even should be.. But at least basic scenarios should now work.

File: lockable/provider_http.py [hunk 4]
```diff
@@ -12,6 +12,17 @@
 MODULE_LOGGER = logging.getLogger('lockable')
 
 
+class RetryWithLogging(Retry):
```
Comment (jupe) [inline lockable/provider_http.py:15]: this was pretty handy solution 👌

**Review by juhhov — APPROVED** [review]

**Pull request merged**